frontend/public/components/operator-lifecycle-manager/index.tsx

```diff
@@ -251,6 +251,8 @@ export const referenceForProvidedAPI = (desc: CRDDescription | APIServiceDefinit
   ? `${(desc as APIServiceDefinition).group}~${desc.version}~${desc.kind}`
   : `${(desc as CRDDescription).name.slice(desc.name.indexOf('.') + 1)}~${desc.version}~${desc.kind}`;
 
+export const referenceForStepResource = (resource: StepResource): GroupVersionKind => `${resource.group || 'core'}~${resource.version}~${resource.kind}`;
```

Comment:
Can we add a utility to build the `group~version~kind` string from separate group/version/kind parameters to `frontend/public/module/k8s/k8s.ts`? We have this sprinkled across the code now... It's error prone building it in so many places, and if we ever need to change the format, it would be difficult.

Not saying to fix this everywhere in this PR, but would be good to add a utility function for new code.

Reply:
I've tried to keep the OLM-specific GVK builders in this module, but the more generic Kube ones should all be in `k8s.ts`, right?

Reply:
Ah, you mean have many functions that parse the three parameters and then one single function to build the string, got it.

Reply:
Right, I'm thinking something as simple as

```js
export const referenceForGroupVersionKind = (group, version, kind) => `${group}~${version}~${kind}`;
```

Then everything that builds one of these strings calls that :)

Reply:
Done